application/backend/app/services/source_service.py

```diff
@@ -2,58 +2,102 @@
 # SPDX-License-Identifier: Apache-2.0
 from uuid import UUID
 
+from sqlalchemy.exc import IntegrityError
 from sqlalchemy.orm import Session
 
+from app.db.schema import SourceDB
+from app.models import Source, SourceType
+from app.models.source import SourceAdapter, SourceConfig
 from app.repositories import SourceRepository
-from app.repositories.base import UniqueConstraintIntegrityError
-from app.schemas import Source
+from app.repositories.base import PrimaryKeyIntegrityError, UniqueConstraintIntegrityError
 
 from .base import (
-    GenericPersistenceService,
+    ResourceInUseError,
     ResourceNotFoundError,
     ResourceType,
+    ResourceWithIdAlreadyExistsError,
     ResourceWithNameAlreadyExistsError,
-    ServiceConfig,
 )
 from .event.event_bus import EventBus, EventType
-from .mappers import SourceMapper
 from .parent_process_guard import parent_process_only
 
 
-class SourceService(GenericPersistenceService[Source, SourceRepository]):
+class SourceService:
     def __init__(self, event_bus: EventBus, db_session: Session):
         self._event_bus: EventBus = event_bus
-        super().__init__(ServiceConfig(SourceRepository, SourceMapper, ResourceType.SOURCE), db_session)
-
-    def get_by_id(self, item_id: UUID) -> Source:
-        source = super().get_by_id(item_id)
-        if not source:
-            raise ResourceNotFoundError(ResourceType.SOURCE, str(item_id))
-        return source
+        self._db_session = db_session
 
     @parent_process_only
-    def create(self, item: Source) -> Source:
+    def create_source(
+        self,
+        name: str,
+        source_type: SourceType,
+        config_data: SourceConfig,
+        source_id: UUID | None = None,
+    ) -> Source:
         try:
-            return super().create(item)
+            db_source = SourceRepository(self._db_session).save(
+                SourceDB(
+                    id=str(source_id) if source_id is not None else None,
+                    name=name,
+                    source_type=source_type,
+                    config_data=config_data.model_dump(mode="json"),
+                )
+            )
+            return SourceAdapter.validate_python(db_source, from_attributes=True)
+        except PrimaryKeyIntegrityError:
+            raise ResourceWithIdAlreadyExistsError(ResourceType.SOURCE, str(source_id))
         except UniqueConstraintIntegrityError:
-            raise ResourceWithNameAlreadyExistsError(ResourceType.SOURCE, item.name)
+            raise ResourceWithNameAlreadyExistsError(ResourceType.SOURCE, name)
 
     @parent_process_only
-    def update(self, source: Source, partial_config: dict) -> Source:
+    def update_source(
+        self,
+        source: Source,
```

Comment:
Is there a reason why we don't pass just `source_id` here similar to `create_source` method?